timemachine/cpp/src/kernels/k_hilbert.cu

```diff
@@ -12,41 +14,43 @@ void __global__ k_coords_to_kv_gather(
     const unsigned int *__restrict__ bin_to_idx,
     unsigned int *__restrict__ keys,
     unsigned int *__restrict__ vals) {
+    int idx = blockIdx.x * blockDim.x + threadIdx.x;
 
-    const int idx = blockIdx.x * blockDim.x + threadIdx.x;
+    const double bx = box[0 * 3 + 0];
+    const double by = box[1 * 3 + 1];
+    const double bz = box[2 * 3 + 2];
 
-    if (idx >= N) {
-        return;
-    }
+    const double inv_bx = 1 / bx;
+    const double inv_by = 1 / by;
+    const double inv_bz = 1 / bz;
 
-    const int atom_idx = atom_idxs[idx];
+    const double inv_bin_width = min(min(inv_bx, inv_by), inv_bz) * (HILBERT_GRID_DIM - 1.0);
 
-    // these coords have to be centered
-    float bx = box[0 * 3 + 0];
-    float by = box[1 * 3 + 1];
-    float bz = box[2 * 3 + 2];
+    while (idx < N) {
```

Comment:
Looks like this also changes the iteration trip count to accommodate for arbitrary launch grid sizes, but we don't actually change the calling side's grid dims at all in this PR?

Reply:
Correct, just slowly adopting this convention as it will make it easier to changes thing in the future without having to do one massive PR.